challenges/rover/framework/ev3robot/robot.py

```diff
@@ -112,14 +110,30 @@ def left(self, angle):
         """
         self.right_motor.speed_sp = round(self.speed / 2)
         self.left_motor.speed_sp = -round(self.speed / 2)
-        self.right_motor.run_forever()
-        self.left_motor.run_forever()
-        moveto = self.gyro_sensor.value() - angle
 
-        while self.gyro_sensor.value() > moveto:
-            # todo check interrupt
-            pass
+        self._rotation_with_exponential_slowdown(angle, operator.gt, operator.sub)
 
+    def _rotation_with_exponential_slowdown(self, angle, comperator, arithmetic_op):
+        """
+        Performs an exponential slow-down during a rotation of the robot.
+        :param target_angle: the angle in degrees
+        :param comparator: custom comparator to check the target_angle vs. the current angle (i.e. the direction of the rotation)
+        :param arithmetic_op: custom arithmetic operator between the current_angle and the target_angle based on the direction of the rotation
+        :return: None
+        """
+        self.right_motor.run_forever()
+        self.left_motor.run_forever()
+        target_angle = arithmetic_op(self.gyro_sensor.value(), angle)
+        while comperator(self.gyro_sensor.value(), target_angle):
+            if min(abs(self.right_motor.speed_sp), abs(self.left_motor.speed_sp)) > self.MIN_ROTATION_SPEED:
+                # use negative values to invert the arithmetic_op
+                # ((-a) + (-b) = - (a + b)
+                partial_target = -(arithmetic_op(-self.gyro_sensor.value(), -target_angle)) * self.ROTATION_EXPONENTIAL_SLOWDOWN_ANGLE_PERCENTAGE
+                while comperator(self.gyro_sensor.value(), partial_target):
+                    pass
+                # TODO: it might be necessary to stop the motor here (?)
+                self.right_motor.speed_sp *= self.ROTATION_EXPONENTIAL_SLOWDOWN_RIGHT_SPEED_PERCENTAGE
+                self.left_motor.speed_sp *= self.ROTATION_EXPONENTIAL_SLOWDOWN_LEFT_SPEED_PERCENTAGE
```

Comment:
We weren't sure whether we need to stop the robot rotation before changing the speed ...